qemu/tests/timerdevice_tsc_enable.py

```diff
@@ -0,0 +1,35 @@
+import logging
+
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    1) Boot a guest with "-cpu host,+invtsc" or "-cpu $cpu_model,+invtsc".
+    2) Check current clocksource and available clocksource and nonstop_tsc flag
+    in guest.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    timeout = float(params.get("login_timeout", 240))
+    session = vm.wait_for_login(timeout=timeout)
+
+    error_context.context("Check current cloclsource", logging.info)
+    cur_clksrc_cmd = params["cur_clksrc_cmd"]
+    current_clksrc = session.cmd_output_safe(cur_clksrc_cmd)
+    avl_clksrc_cmd = params["avl_clksrc_cmd"]
+    avl_clksrc = session.cmd_output_safe(avl_clksrc_cmd)
+    check_tsc_flag_cmd = params["check_tsc_flag_cmd"]
+    tsc_flag = session.cmd_output_safe(check_tsc_flag_cmd)
+
+    if "tsc" not in current_clksrc or "nonstop_tsc" not in tsc_flag or \
+       "tsc kvm-clock acpi_pm" not in avl_clksrc:
+        test.fail("Current clocksource, tsc flag and available clocksource are "
+                  "respectively %s, %s and %s" % (current_clksrc, tsc_flag,
+                                                  avl_clksrc))
```

Comment:
How about split it ?  Make it more easy to know which scenario it failed with.
something like:
`Current clocksource is ***, expect clocksource is ***`